File: chapter 5/transactions.rb
require 'redis'
@redis = Redis.new(:host => "127.0.0.1", :port => 6379)

@redis.multi do |multi|
  multi.set "transaction:key", "A string in a transactional block"
  multi.incr "transaction:counter"
end

@redis.get "transaction:key"
# => "A string in a transactional block"

File: chapter 5/transactions.py
import redis
client = redis.StrictRedis(host='localhost', port=6379)

with client.pipeline(transaction=True) as transaction:
   transaction.set('transaction:key', 'A string in a transactional block')
   transaction.incr('transaction:counter')
   transaction.get('transaction:key')
   result = transaction.execute()
   # [True, 2, 'A string in a transactional block']
